File: VehicleMonitoringSystemWeb/VehicleMonitoringSystemServer/VehicleMonitoringSystemServer/Data/Models/VehicleData.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VehicleMonitoringSystemServer.Data.Models
{
    [Table("vehicle_data")]
    public class VehicleData
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }
        
        [Column("vehicle_id")]
        public int VehicleId { get; set; }
        [ForeignKey("VehicleId")]
        public Vehicle Vehicle { get; set; }

        [Column("date")] 
        public DateTime Date { get; set; }
        
        [Column("latitude")] 
        public decimal Latitude { get; set; }
        
        [Column("longitude")] 
        public decimal Longitude { get; set; }
        
        [Column("fuel_lever")] 
        public decimal FuelLevel { get; set; }
    }
}

File: VehicleMonitoringSystemWeb/vms_frontend/src/models/VehicleData.ts
import Vehicle from "./Vehicle";

export default class VehicleData {
  public id: number | undefined;
  public vehicle_id: number | undefined;
  public vehicle: Vehicle | undefined;
  public user_id: number | undefined;
  public datetime: Date | undefined;
  public latitude: string | undefined;
  public longitude: string | undefined;

  // public toJSON() {
  //   return {
  //     Id: this.id, VehicleId: this.vehicleId, UserId: this.userId,
  //     Datetime: this.datetime, Latitude: this.latitude, Longitude: this.longitude
  //   }
  // }
}
